tests/test_api_lib_policy.py

```diff
@@ -3022,14 +3022,49 @@ def test_36_init_registrationcode_length_contents(self):
                    action="{0!s}={1!s}".format(ACTION.REGISTRATIONCODE_CONTENTS, "+n"))
         # request, that matches the policy
         req.all_data = {"user": "cornelius", "realm": "home", "type": "registration"}
-        init_registrationcode_length_contents(req)
+        init_token_length_contents(req)
         # Check, if the tokenlabel was added
         self.assertEqual(req.all_data.get("registration.length"), "6")
         self.assertEqual(req.all_data.get("registration.contents"), "+n")
         # delete policy
         delete_policy("reg_length")
         delete_policy("reg_contents")
 
+    def test_37_init_password_length_contents(self):
+        g.logged_in_user = {"username": "admin1",
+                            "realm": "",
+                            "role": "admin"}
+        builder = EnvironBuilder(method='POST',
+                                 data={'type': "password", "genkey": 1},
+                                 headers={})
+        env = builder.get_environ()
+        req = Request(env)
+
+        # first test without any policy
+        req.all_data = {"user": "cornelius", "realm": "home", "type": "pw", "genkey": 1}
+        init_token_length_contents(req)
+        # Check, if the tokenlabel was added
+        from privacyidea.lib.tokens.passwordtoken import DEFAULT_LENGTH, DEFAULT_CONTENTS
+        self.assertEqual(req.all_data.get("pw.length"), DEFAULT_LENGTH)
+        self.assertEqual(req.all_data.get("pw.contents"), DEFAULT_CONTENTS)
+
+        # now create a policy for the length of the registration code
+        set_policy(name="pw_length",
+                   scope=SCOPE.ENROLL,
+                   action="{0!s}={1!s}".format(ACTION.PASSWORD_LENGTH, 6))
+        set_policy(name="pw_contents",
+                   scope=SCOPE.ENROLL,
+                   action="{0!s}={1!s}".format(ACTION.PASSWORD_CONTENTS, "+n"))
+        # request, that matches the policy
+        req.all_data = {"user": "cornelius", "realm": "home", "type": "pw"}
+        init_token_length_contents(req)
+        # Check, if the tokenlabel was added
+        self.assertEqual(req.all_data.get("pw.length"), "6")
+        self.assertEqual(req.all_data.get("pw.contents"), "+n")
+        # delete policy
+        delete_policy("pw_length")
+        delete_policy("pw_contents")
+
```

Comment:
Please also add a test that if the token is neither `pw` nor `registration` we don't have the `lenght` and `content` parameter in `all_data`

Reply:
Will check and will do.

Reply:
Done